cms/static/js/factories/import.js

```diff
@@ -1,30 +1,48 @@
 define([
-    'js/views/import', 'jquery', 'gettext', 'jquery.fileupload', 'jquery.cookie'
-], function(Import, $, gettext) {
+    'domReady', 'js/views/import', 'jquery', 'gettext', 'jquery.fileupload', 'jquery.cookie'
+], function(domReady, Import, $, gettext) {
+
     'use strict';
+
     return function (feedbackUrl, library) {
-        var dbError;
-        if (library) {
-            dbError = gettext('There was an error while importing the new library to our database.');
-        } else {
-            dbError = gettext('There was an error while importing the new course to our database.');
-        }
         var bar = $('.progress-bar'),
             fill = $('.progress-fill'),
             submitBtn = $('.submit-button'),
-            chooseBtn = $('.choose-file-button'),
+            chooseBtn = $('.view-import .choose-file-button'),
             defaults = [
                 gettext('There was an error during the upload process.') + '\n',
                 gettext('There was an error while unpacking the file.') + '\n',
                 gettext('There was an error while verifying the file you submitted.') + '\n',
                 dbError + '\n'
             ],
-            // Display the status of last file upload on page load
-            lastFileUpload = $.cookie('lastfileupload'),
+            previousImport = Import.storedImport(),
+            dbError,
             file;
 
-        if (lastFileUpload){
-            Import.getAndStartUploadFeedback(feedbackUrl.replace('fillerName', lastFileUpload), lastFileUpload);
+        var onComplete = function () {
+            bar.hide();
+            chooseBtn
+                .find('.copy').html(gettext("Choose new file")).end()
+                .show();
+        }
+
+        if (library) {
+            dbError = gettext('There was an error while importing the new library to our database.');
+        } else {
+            dbError = gettext('There was an error while importing the new course to our database.');
+        }
```

Comment:
@AkA84 dbError is only used in defaults array declaration above, so this block should stay where it was.